core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -569,28 +569,34 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
         return result;
     }
 
+    public Engine.NoOp prepareMarkingSeqNoAsNoOp(long seqNo, String reason) {
```

Comment:
To follow the standard that we have use here, I think this should be named `prepareNoOpOnReplica(...)` (since we have `prepareDeleteOnReplica` and `prepareIndexOnReplica`)